packages/envd/internal/utils/rfsnotify.go

```diff
@@ -0,0 +1,12 @@
+package utils
+
+import "path/filepath"
+
+// FsnotifyPath creates an optionally recursive path for fsnotify/fsnotify internal implementation
+func FsnotifyPath(path string, recursive bool) string {
```

Comment:
❤️ this, i was initially turned off by the fact that the original code hardcoded the branch to never allow to use the `"..."` path with a cryptic comment saying "allowed only for test". The tests you wrote seem to show that this works, not necessarily a request for code change but what would you do to test/find proof that this doesn't listen to  special linux files (Block file, Character device file, Symbolic link file, Socket file, etc) , also, i the same vein, what root folders would you exclude, if any, to avoid spamming?

Reply:
This is a very good point.

**Special Linux files**
Basic files should be handled the same as without recursive watch. The implementation of recursion uses either isDir() or isDir := mask&unix.IN_ISDIR == unix.IN_ISDIR to watch nested directories (based on the source code). I've tested block files, device files (/dev), socket files (using python), etc. and I didn't find any issues (excessive updates or crashing). Symbolic links of folders are followed.

**Spam**
About spamming, maybe it would make sense to prevent users for listening on certain directories (/dev, /proc, /var, /sys). But, even without the recursive watch, they can watch these folders or too many files and break the containers. So it might not be necessarily an issue.

Reply:
Cool thanks for checking the behaviour for special files 👍 
For the spamming I was thinking of not allowing those folders to be watched at all, sometimes a user may want to watch root `/` folder and with this we could lower the default linux files being watched from ~4000 to ~800 last i checked. Not needed for this PR tho, just wanted to see your thought process, thanks.

Reply:
Ok 👍 I agree with the follow-up PR